app/api/chat/agents/route.ts

```diff
@@ -26,23 +26,28 @@ const convertVercelMessageToLangChainMessage = (message: VercelChatMessage) => {
 };
 
 const convertLangChainMessageToVercelMessage = (message: BaseMessage) => {
-  if (message._getType() === "human") {
+  if (message.getType() === "human") {
     return { content: message.content, role: "user" };
-  } else if (message._getType() === "ai") {
+  } else if (message.getType() === "ai") {
     return {
       content: message.content,
       role: "assistant",
-      tool_calls: (message as AIMessage).tool_calls,
+      parts: (message as AIMessage).tool_calls,
+    };
+  } else if (message.getType() === "tool") {
+    return {
+      content: message.content,
+      role: "system",
```

Comment:
I think this should be `role: tool`?
```suggestion
      role: "tool",
```

Reply:
Vercel `Message` has bound the type to `role: 'system' | 'user' | 'assistant' | 'data';` so I thought system was the closest to tool

Reply:
hmm weird. id need to read the docs then but i dont think system would be the right role for this